pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -71,7 +71,7 @@ func (s *Service) ReconcileLoadBalancer(clusterName string, openStackCluster *in
 		}
 	}
 
-	fp, err := s.networkingService.GetOrCreateFloatingIP(openStackCluster, openStackCluster.Spec.ControlPlaneEndpoint.Host)
+	fp, err := s.networkingService.GetOrCreateFloatingIP(openStackCluster, openStackCluster.Spec.APIServerFloatingIP)
```

Comment:
commit 4d3d98c1f77393d776c0278ec97dc877dfb775b6
Author: Hidekazu Nakamura <hidekazuna@gmail.com>
Date:   Fri Dec 11 07:32:54 2020 +0000

    Refactor loadbalancer package

    This commit removes floating IP related code from loadbalancer package.


@hidekazuna can you help to check this proposed change?

Reply:
We need to consider the case `openStackCluster.Spec.APIServerFloatingIP` is not set.

Current flow:
If `openStackCluster.Spec.APIServerFloatingIP` is not set, create floating IP and set `openStackCluster.Spec.ControlPlaneEndpoint.Host`. After that we create load balancer using `openStackCluster.Spec.ControlPlaneEndpoint.Host`.

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/controllers/openstackcluster_controller.go#L407-L435

Reply:
Is it time to introduce openStackCluster.Status.APIServerFloatingIP?

Reply:
I added a new commit with fallback to original behavior.

> introduce openStackCluster.Status.APIServerFloatingIP

I'm not sure what you mean, I'm not super familiar with the codebase, so a little hesitant with doing bigger changes. However, if guided, I should be able to do a proper update.

Reply:
@jsen- I think you're current version is a fine solution for now. We can always adjust the inner workings in a follow-up PR

@hidekazuna I would probably want to take a look how other providers are handling this. Although I think we shouldn't change a lot here. There is a loadbalancer provider upcoming with v1alpha4 which imho we should implement. This will change the code quite a bit (although I assume the behaviour should stay the same)

Reply:
For reference:
* current google doc version of the PR: https://docs.google.com/document/d/1wJrtd3hgVrUnZsdHDXQLXmZE3cbXVB5KChqmNusBGpE/edit
* but it was mentioned today in Slack that it will be converted to a PR in the cluster-api repo asap (If I got it correctly there's a deadline on Monday for v1alpha4 changes)

Reply:
@sbueringer Other providers does not support non lb case. We will re-visit this issue after v1alpha4 implementation merged.

Reply:
@hidekazuna Yup we will have to figure that out with v1alpha4 & the lb provider. But should be possibel as the lb provider itself is optional. So we should be able to have `cluster-tempate` with the lb provider and a `cluster-template-without-lb` without the lb provider